presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisySumAvgState.java

```diff
@@ -22,12 +22,14 @@
 import static io.airlift.slice.SizeOf.SIZE_OF_DOUBLE;
 
 /**
- * State class for noisy sum aggregation.
- * This class extends noisy count state to store the number of input rows explicitly
- * so that NULL can be returned if there is no input rows.
+ * State class for noisy sum and noisy avg aggregations.
+ * This class extends noisy count state to store the number of input rows explicitly.
+ * That count is used to compute noisy average.
+ * In the case of noisy sum, if there is no input rows, NULL should be returned
+ * and that count is used to check that condition.
  */
-@AccumulatorStateMetadata(stateSerializerClass = NoisySumStateSerializer.class)
-public interface NoisySumState
+@AccumulatorStateMetadata(stateSerializerClass = NoisySumAvgStateSerializer.class)
+public interface NoisySumAvgState
```

Comment:
nit: I realize this is an annoying comment, and I don't have strong feelings on this, but `NoisySumAndCountState` may be slightly more in line with other state classes (e.g., https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/LongAndDoubleState.java) and would make it more obvious why it extends `NoisyCountState`.

Reply:
Agree. I am working on count_if as well and there is a whole bunch of refactoring as well. I actually use NoisySumAvgState as well. But NoisyCountAndSumState does sound better